### PR TITLE
Support linking CommonJS projects

### LINKED ISSUE #634: Scala.js frontend instructions
Hello, I'm trying to get bloop running with Slinky's example project (https://slinky.shadaj.me/docs/installation/) to avoid the SBT overhead

I figured to get hot reloading I'll need to run webpack in a separate terminal, but looking at the docs I can't seem to figure out how to run `fastOptJS` in bloop.

I tried `bloop link my-app` and got

```
[E] No main classes were found in the project 'my-app'
```

which I think is due to slinky using a JS file as the main file (to perform webpack hot reloading):

```
require("./my-app-fastopt.js").entrypoint.main();

if (module.hot) {
    module.hot.accept();
}
```

So I guess the question then becomes: Is there a way to run `fastOptJS` by itself?

### PR DESCRIPTION
**Pull request by tindzk**
CommonJS projects do not require a main class.

Closes #634.

### CODE REVIEW

File: frontend/src/main/scala/bloop/engine/Interpreter.scala [hunk 1]
```diff
@@ -308,22 +309,32 @@ object Interpreter {
...
+                LinkTask.linkMainWithNative(cmd, project, state, mainClass, target, platform)
             }
+
+          case platform @ Platform.Js(config, _, _) =>
+            val target = ScalaJsToolchain.linkTargetFrom(project, config)
+
+            // A CommonJS module does not have an entry point
+            if (config.kind == Config.ModuleKindJS.CommonJSModule)
+              LinkTask.linkMainWithJs(cmd, project, state, None, target, platform)
+            else
```
Comment (jvican) [inline frontend/src/main/scala/bloop/engine/Interpreter.scala]:
```suggestion
            else {
              getMainClass(state, project, cmd.main) match {
                   case Left(state) => Task.now(state)
                   case Right(mainClass) =>
                     LinkTask.linkMainWithJs(cmd, project, state, Some(mainClass), target, platform)
                 }
               }
```

Comment (tindzk) [reply frontend/src/main/scala/bloop/engine/Interpreter.scala]: Sure, I will add a test.

**Comment by jvican** [comment]
I've done quite a bit of progress here, will report back tomorrow

**Comment by jvican** [comment]
@tindzk Closing this one for now to keep the PR queue clean, I'll reopen when I see myself with the bandwidth to make progress on the Scala.js side. Right now I'm still ironing out compiler performance.

**Pull request closed without merging**